### PR TITLE
Updating getobjectsbyobjectId call for objectIds > 1000

### PR DESCRIPTION
**Pull request by darshanhs90**
## Description
Updating getobjectsbyobjectId call for objectIds > 1000,since they API limits only 1000 objectIds

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/ActiveDirectoryClient.cs [hunk 1]
```diff
@@ -251,8 +251,17 @@ public List<PSADObject> ListUserGroups(string principal)
         public List<PSADObject> GetObjectsByObjectId(List<string> objectIds)
         {
             List<PSADObject> result = new List<PSADObject>();
-            var adObjects = GraphClient.Objects.GetObjectsByObjectIds(new GetObjectsParameters { ObjectIds = objectIds, IncludeDirectoryObjectReferences = true });
-            result.AddRange(adObjects.Select(o => o.ToPSADObject()));
+            IPage<AADObject> adObjects;
+            for(int i=0; i<objectIds.Count; i+=1000)
+            {
+                if((i+1000) > objectIds.Count){
+                    adObjects = GraphClient.Objects.GetObjectsByObjectIds(new GetObjectsParameters { ObjectIds = objectIds.GetRange(i, objectIds.Count - i), IncludeDirectoryObjectReferences = true });
```
Comment (wisexp) [inline src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/ActiveDirectoryClient.cs]: >objectIds.Count - i [](start = 135, length = 19)

compute the end index in "if-else" block then merge two graph calls into one line

File: src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/ActiveDirectoryClient.cs [hunk 2]
```diff
@@ -251,8 +251,17 @@ public List<PSADObject> ListUserGroups(string principal)
         public List<PSADObject> GetObjectsByObjectId(List<string> objectIds)
         {
             List<PSADObject> result = new List<PSADObject>();
-            var adObjects = GraphClient.Objects.GetObjectsByObjectIds(new GetObjectsParameters { ObjectIds = objectIds, IncludeDirectoryObjectReferences = true });
-            result.AddRange(adObjects.Select(o => o.ToPSADObject()));
+            IPage<AADObject> adObjects;
+            for(int i=0; i<objectIds.Count; i+=1000)
```
Comment (wisexp) [inline src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/ActiveDirectoryClient.cs]: >1000 [](start = 47, length = 4)

it will be better to write a comment that graph only supports up to 1000 ids.

File: src/ResourceManager/Resources/Commands.Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ResourceGroupTests/TestGetNonExistingResourceGroup.json [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
         ],
         "accept-language": [
           "en-US"
         ],
         "User-Agent": [
-          "Microsoft.Azure.Management.ResourceManager.ResourceManagementClient/1.1.5-preview"
+          "FxVersion/4.7.2046.0",
+          "OSName/Windows_10_Enterprise",
+          "OSVersion/6.3.15063",
+          "Microsoft.Azure.Management.ResourceManager.ResourceManagementClient/1.6.0-preview"
```
Comment (wisexp) [inline src/ResourceManager/Resources/Commands.Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ResourceGroupTests/TestGetNonExistingResourceGroup.json]: >entClient/1.6.0-preview [](start = 69, length = 23)

are these stuff related to your change?

Comment (darshanhs90) [inline src/ResourceManager/Resources/Commands.Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ResourceGroupTests/TestGetNonExistingResourceGroup.json]: I just ran the other test cases to make sure they still work in record mode since the session records had become stale( as the test cases were ran almost a year back)

---
In reply to: [117386585](https://github.com/Azure/azure-powershell/pull/4002#discussion_r117386585) [](ancestors = 117386585)

Comment (wisexp) [inline src/ResourceManager/Resources/Commands.Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ResourceGroupTests/TestGetNonExistingResourceGroup.json]: I think it's better to update them in a separate PR for tracking purpose.

---
In reply to: [117542183](https://github.com/Azure/azure-powershell/pull/4002#discussion_r117542183) [](ancestors = 117542183,117386585)

Comment (darshanhs90) [inline src/ResourceManager/Resources/Commands.Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.ResourceGroupTests/TestGetNonExistingResourceGroup.json]: reverted the extra ones

---
In reply to: [117567000](https://github.com/Azure/azure-powershell/pull/4002#discussion_r117567000) [](ancestors = 117567000,117542183,117386585)

**Comment by wisexp** [comment]
:shipit:

**Comment by darshanhs90** [comment]
@shahabhijeet Is there anything blocking from my side for the pr to be merged?

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1605/ 

signing run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1066/

LGTM once these pass

**Comment by darshanhs90** [comment]
@markcowl The signing run is failing for 'Stop-AzureStorageBlobCopy' return type 'AzureStorageBlob'."
ERROR - System.InvalidOperationException: One or more errors occurred in validation. See the analysis repots at D:\workspace\powershell-sign\src\Package for details at StaticAnalysis.AnalysisLogger.CheckForIssues(Int32 maxSeverity) in D:\workspace\powershell-sign\tools\StaticAnalysis\Loggers\AnalysisLogger.cs:line 185

do let me know no what can be done to get this PR merged

**Comment by markcowl** [comment]
@darshanhs90 This is failing due to reports of breaking change issues.  Most likely this means you are not up to date with the preview branch.  Please pull the latest from the preview branch and push to your fork.

**Comment by darshanhs90** [comment]
@markcowl updated the code with the latest code from the upstream

**Comment by markcowl** [comment]
new on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1612/

new signing run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1082/

**Comment by darshanhs90** [comment]
the ondemand run passed,however the signing run failed again even after i restarted it
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/1083/
